### ISSUE TITLE
OCP\Files\NotFoundException but the file seems to exist

### ISSUE THREAD

**Issue by Zibc** [nextcloud/server#710]
Good morning,

Last week end I upgraded my owncloud to nextcloud, everything went like a breeze during the update. Well almost everything.

My log file is now populated with a massaive amount of exception, either when somebody connects to the cloud or when a cron job runs.
It does not seems to bother the installation a lot because everything works.

the stack trace is the following:

Exception:OCP\Files\NotFoundException
Message:File with id 1299 has not been found.
Code:0
Trace:
0 /path/to/my/cloud/apps/files_sharing/lib/sharedstorage.php(90): OC\Files\View->getPath(1299)
1 /path/to/my/cloud/apps/files_sharing/lib/sharedstorage.php(581): OC\Files\Storage\Shared->init()
2 /path/to/my/cloud/lib/private/files/storage/wrapper/wrapper.php(381): OC\Files\Storage\Shared->getCache('', Object(OCA\Files_Trashbin\Storage))
3 /path/to/my/cloud/lib/private/files/cache/scanner.php(88): OC\Files\Storage\Wrapper\Wrapper->getCache()
4 /path/to/my/cloud/apps/files_sharing/lib/sharedstorage.php(595): OC\Files\Cache\Scanner->__construct(Object(OCA\Files_Trashbin\Storage))
5 /path/to/my/cloud/lib/private/files/storage/wrapper/wrapper.php(395): OC\Files\Storage\Shared->getScanner('', Object(OCA\Files_Trashbin\Storage))
6 /path/to/my/cloud/lib/private/files/utils/scanner.php(127): OC\Files\Storage\Wrapper\Wrapper->getScanner()
7 /path/to/my/cloud/apps/files/lib/backgroundjob/scanfiles.php(87): OC\Files\Utils\Scanner->backgroundScan('')
8 /path/to/my/cloud/apps/files/lib/backgroundjob/scanfiles.php(111): OCA\Files\BackgroundJob\ScanFiles->runScanner(Object(OC\User\User))
9 /path/to/my/cloud/lib/private/backgroundjob/job.php(52): OCA\Files\BackgroundJob\ScanFiles->run(NULL)
10 /path/to/my/cloud/lib/private/backgroundjob/timedjob.php(53): OC\BackgroundJob\Job->execute(Object(OC\BackgroundJob\JobList), Object(OC\Log))
11 /path/to/my/cloud/cron.php(145): OC\BackgroundJob\TimedJob->execute(Object(OC\BackgroundJob\JobList), Object(OC\Log))
12 {main}
File:/path/to/my/cloud/lib/private/files/view.php
Line:1677

I checked in the database what file 1299 was and the file exists and is a shared file. If I remove the share, the exception still happens.
There is also a reference to this file in the oc_mounts table.
### Steps to reproduce

Well, I am not sure how to actually reproduce this issue ... 
### Expected behaviour

No exception in the log file
### Actual behaviour

Log file filed with this exception
### Server configuration

**Operating system**: Debian 8

**Web server:** Apache 2.4

**Database:** MySQL

**PHP version:** 5.6

**Nextcloud version:** 9.0.53

**Updated from an older Nextcloud/ownCloud or fresh install:** updated from owncloud

**Where did you install Nextcloud from:** from the source

**Signing status:** no erros have been found

**List of activated apps:**
<details>
<summary>

App list</summary>

Enabled:
- admin_audit: 1.0.0
- bookmarks: 0.666666
- calendar: 1.3.1
- comments: 0.2
- contacts: 1.3.1.0
- dav: 0.1.6
- federatedfilesharing: 0.1.0
- federation: 0.0.4
- files: 1.4.4
- files_pdfviewer: 0.8.1
- files_sharing: 0.9.1
- files_texteditor: 2.1
- files_trashbin: 0.8.0
- files_versions: 1.2.0
- files_videoplayer: 0.9.8
- gallery: 14.5.0
- mail: 0.5.2
- news: 8.8.0
- notifications: 0.2.3
- password_policy: 1.0.0
- provisioning_api: 0.4.1
- systemtags: 0.2
- templateeditor: 0.1
- theming: 0.1.0
- updatenotification: 0.1.0
  Disabled:
- activity
- encryption
- external
- files_external
- firstrunwizard
- user_external
- user_ldap
- user_saml
  </details>

**The content of config/config.php:**
<details>
<summary>

Config report</summary>

```
{
    "system": {
        "instanceid": "51b47554b7363",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "/path/to/my/data",
        "dbtype": "mysql",
        "version": "9.0.53.0",
        "dbname": "cloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "loglevel": 2,
        "theme": "",
        "maintenance": false,
        "trusted_domains": [
            "my.domain.tld"
        ],
        "share_folder": "\/Shared",
        "appcodechecker": false,
        "secret": "***REMOVED SENSITIVE VALUE***",
        "log_rotate_size": 2097152,
        "trashbin_retention_obligation": "auto",
        "memcache.local": "\\OC\\Memcache\\APCu"
    }
}
```

</details>

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no
### Client configuration

**Browser:** firefox

**Operating system:** linux mint 17.3
#### Web server error log

<details>
<summary>

Web server error log</summary>

```

```

</details>
#### Nextcloud log (data/nextcloud.log)

<details>
<summary>

Nextcloud log</summary>

```
```

</details>
#### Browser log

<details>
<summary>

Browser log</summary>

```
```

</details>

**Comment by rullzer** [comment]
It could be that this is fixed with #1013 as @derkostka mentions

**Comment by blizzz** [comment]
Closing as possibly fixed, since we got no further activity here.
